iam/snippets/roles_undelete.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START iam_undelete_role]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"golang.org/x/oauth2/google"
+	iam "google.golang.org/api/iam/v1"
+)
+
+// undeleteRole restores a deleted custom role.
+func undeleteRole(w io.Writer, name string, projectID string) (*iam.Role, error) {
```

Comment:
Might want to put `projectID` first and document what `name` is. Should it be `fullResourceName` (and the code updated)?

Reply:
Swapped vars. Throughout the docs, we use "name" when we mean local name, and "fullResourceName" when the whole thing is required. I don't think this will confuse customers.